src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -3350,26 +3350,21 @@ void walkInnerSymbols(ObjectType type, String innerNamespace) {
         String qualifiedName = innerNamespace + '.' + propName;
         if (provides.contains(qualifiedName)) continue;
         Node node = innerProps.get(namedType);
+        // Start namespace if this is the first pair being processed.
+        if (!foundNamespaceMembers &&
+            (pType.isEnumType() || isClassLike(pType) || isTypedef(pType))) {
```

Comment:
This is fragile because the clauses here have to match the if stmts below. At least add a comment to that effect.

Reply:
Changed it back. Couldn't think of nice way to reuse the code and prevent possible issue that you mentioned. So decided not worth it :)